nix/build.nix

```diff
@@ -0,0 +1,18 @@
+{
+  base,
+}:
+let
+  niv = import ./sources.nix;
+  nixpkgsSrc = niv.nixpkgs;
+
+  nixpkgs = import niv.nixpkgs {};
+  overrides = self: super: {
+    polysemy = self.callCabal2nix "polysemy" base {};
+    polysemy-plugin = self.callCabal2nix "polysemy-plugin" "${base}/polysemy-plugin" {};
+  };
+  ghc = nixpkgs.haskell.packages.ghc8102.override { inherit overrides; };
+in {
+  inherit ghc;
```

Comment:
I would call this `haskellPackages` instead of `ghc`, because that's the set of Haskell packages for GHC 8.10.2, not the GHC package itself.

And with that in mind, I'm not sure what the value of `inherit ghc;` on line 15 is?

Reply:
I named it `ghc` because I'm used to having both that and `ghcjs` from obelisk projects; in this case it doesn't make much sense, I agree.

The purpose of `inherit ghc` is to expose the overrides, for example for use in `shell.nix`. So far there's nothing in there, so it doesn't provide much utility, but I'd say it's still convenient for manual use.

Reply:
Ah I see